src/modules/kvs-watch/kvs-watch.c

```diff
@@ -556,13 +593,18 @@ static int handle_append_response (flux_t *h,
                 errno = EPROTO;
                 goto error_respond;
             }
+            if (w->flags & FLUX_KVS_STREAM)
+                goto out;
             new_end_index = treeobj_get_count (val) - 1;
             if (new_end_index > w->prev_end_index) {
                 w->prev_start_index = w->prev_end_index + 1;
                 w->prev_end_index = new_end_index;
             }
             else if (new_end_index < w->prev_end_index) {
-                errprintf (&err, "key watched with WATCH_APPEND shortened");
+                if (w->flags & FLUX_KVS_WATCH_APPEND)
+                    errprintf (&err, "key watched with WATCH_APPEND shortened");
+                else
+                    errprintf (&err, "key being streamed shortened");
```

Comment:
Hmm, seems like in the STREAM case, we should just be working from a static version of the treeobj?

Reply:
oh yeah that's right.  I was just trying to avoid the "WATCH_APPEND" in an error message.

Reply:
Maybe just make it
```c
// cannot occur with FLUX_KVS_STREAM
errprintf (&err, "key watched with WATCH_APPEND shortened");
```
if that makes sense?  Otherwise the error message implies that it's an expected case which is a bit confusing.